openstf/model/metamodels/missing_values_handler.py

```diff
@@ -0,0 +1,120 @@
+
+    imputation_strategy: str, default=None
+        Imputing strategy for the imputer. If None, nan values are not allow and the linear regression will raise a ValueError.
+
+    fill_value: optional
+        Value used by the imputer for missing values.
+
+     Attributes
+    ----------
+        feature_names: list(str)
+            All input feature.
+
+        non_null_columns_: list(str)
+            Valid features used by the regressor.
+
+        n_features_in_: int
+            Number of input features.
+
+        regressor_: RegressorMixin
+            Regressor fitted on valid columns.
+
+        imputer_: SimpleImputer
+            Imputer for missig value fitted on valid columns.
+
+        pipeline_: Pipeline
+            Pipeline that chains the imputer and the regressor.
+
+        feature_importances_: ndarray (n_features_in_, )
+            The feature importances from the regressor for valid features and zero otherwise.
+    """
+
+    def __init__(
+        self,
+        base_estimator,
+        missing_values=np.nan,
+        imputation_strategy=None,
+        fill_value=None,
+    ):
+        self.base_estimator = base_estimator
+        self.missing_values = missing_values
+        self.imputation_strategy = imputation_strategy
+        self.fill_value = fill_value
+
+    def _get_tags(self):
+        tags = self.base_estimator._get_tags()
+        tags["multioutput"] = False
+        tags["allow_nan"] = self.imputation_strategy is not None
+        return tags
+
+    def fit(self, X, y):
+
+        _, y = check_X_y(X, y, force_all_finite="allow-nan", y_numeric=True)
+        if type(X) != pd.DataFrame:
+            X = pd.DataFrame(np.asarray(X))
+
+        # Remove always null columns
+        columns = X.isnull().all(0)
+        self.feature_names_ = list(X.columns)
+        self.non_null_columns_ = list(columns[~columns].index)
+        self.n_features_in_ = X.shape[1]
+
+        self.regressor_ = clone(self.base_estimator)
+        if self.imputation_strategy is None:
+            self.imputer_ = None
+            self.pipeline_ = Pipeline([("regressor", self.regressor_)])
+        else:
+            self.imputer_ = SimpleImputer(
+                missing_values=self.missing_values,
+                strategy=self.imputation_strategy,
+                fill_value=self.fill_value,
+            )
+            self.pipeline_ = Pipeline(
+                [("imputer", self.imputer_), ("regressor", self.regressor_)]
+            )
+
+        self.pipeline_.fit(X[self.non_null_columns_], y)
+
+        if hasattr(self.regressor_, "feature_importances_"):
+            reg_feature_importances = self.regressor_.feature_importances_
+            self.feature_importances_ = np.zeros(self.n_features_in_)
+            j = 0
+            for i, c in enumerate(self.feature_names_):
+                if c in self.non_null_columns_:
+                    self.feature_importances_[i] = reg_feature_importances[j]
+                    j += 1
```

Comment:
Can we make this more pythonic?
I'm not exactly sure what you want to do here, but I'm guessing that you try to only keep the feature importances of the non_null_columns_?
In that case, would this work:
`self.feature_importances_ = [x for x in self.regressor_.feature_importances if x not in self.non_null_columns_]` ?

Reply:
Your proposal would not work. However a possible more pythonic way would be:
```python
reg_feature_importances_dict = dict(zip(self.non_null_columns, reg_feature_importances)
self.feature_importances_ = np.array([reg_feature_importances_dict.get(c, 0) for c in self.feature_names_])
```